internal/creds.go

```diff
@@ -11,13 +11,25 @@ import (
 	"io/ioutil"
 
 	"golang.org/x/oauth2"
+	"google.golang.org/api/internal/impersonate"
 
 	"golang.org/x/oauth2/google"
 )
 
 // Creds returns credential information obtained from DialSettings, or if none, then
 // it returns default credential information.
 func Creds(ctx context.Context, ds *DialSettings) (*google.Credentials, error) {
+	creds, err := creds(ctx, ds)
```

Comment:
rename unexported "creds" to "baseCreds" perhaps? indicating that it doesn't cover impersonation.

Reply:
Ack.